test/suites/basic.sh

```diff
@@ -547,7 +548,17 @@ test_basic_usage() {
   ! lxc profile rename default foobar || false
   ! lxc profile delete default || false
 
+  # Should fail to override root device storage pool when container is already initialised.
   lxc init testimage c1
   ! lxc config device override c1 root pool bla || false
   lxc rm -f c1
+
+  # Should fail to override root device storage pool when the new pool does not exist.
+  ! lxc init testimage c1 -d root,pool=bla || false
+
+  # Should succeed in overriding root device storage pool when the pool does exist and the override occurs at create time.
+  lxc storage create bla dir
+  lxc init testimage c1 -d root,pool=bla
```

Comment:
Can you add a check for `lxc config show <instance> --expanded` here to check the pool override took effect in the expanded config?

Reply:
Done.